ext/canvas/01_image.js

```diff
@@ -204,147 +209,143 @@ function createImageBitmap(
+        "options.resizeHeight has to be greater than 0",
         "InvalidStateError",
       ),
     );
   }
 
   const imageBitmap = webidl.createBranded(ImageBitmap);
 
-  if (ObjectPrototypeIsPrototypeOf(ImageDataPrototype, image)) {
-    const processedImage = processImage(
-      image[_data],
-      image[_width],
-      image[_height],
-      sxOrOptions,
-      sy,
-      sw,
-      sh,
-      options,
+  // 3.
+  const isBlob = ObjectPrototypeIsPrototypeOf(BlobPrototype, image);
+  const isImageData = ObjectPrototypeIsPrototypeOf(ImageDataPrototype, image);
+  if (!isBlob && !isImageData) {
+    return PromiseReject(
+      new DOMException(
+        `${prefix}: The provided value for 'image' is not of type '(${
+          ArrayPrototypeJoin(imageBitmapSources, " or ")
+        })'.`,
+        "InvalidStateError",
+      ),
     );
-    imageBitmap[_bitmapData] = processedImage.data;
-    imageBitmap[_width] = processedImage.outputWidth;
-    imageBitmap[_height] = processedImage.outputHeight;
-    return PromiseResolve(imageBitmap);
   }
-  if (ObjectPrototypeIsPrototypeOf(BlobPrototype, image)) {
-    return (async () => {
-      const data = await image.arrayBuffer();
-      const mimetype = sniffImage(image.type);
-      if (mimetype !== "image/png") {
-        throw new DOMException(
-          `Unsupported type '${image.type}'`,
-          "InvalidStateError",
+
+  // 4.
+  return (async () => {
+    //
+    // For performance reasons, the arguments passed to op are represented as numbers that don't need to be serialized.
+    //
+
+    let width = 0;
+    let height = 0;
+    // If the of image doesn't have a MIME type, mark it as 0.
+    let mimeType = 0;
+    let imageBitmapSource, buf;
+    if (isBlob) {
+      imageBitmapSource = 0;
+      buf = new Uint8Array(await image.arrayBuffer());
+      const mimeTypeString = sniffImage(image.type);
+
+      if (mimeTypeString === "image/png") {
+        mimeType = 1;
+      } else if (mimeTypeString === "image/jpeg") {
+        mimeType = 2;
+      } else if (mimeTypeString === "image/gif") {
+        mimeType = 3;
+      } else if (mimeTypeString === "image/bmp") {
+        mimeType = 4;
+      } else if (mimeTypeString === "image/x-icon") {
+        mimeType = 5;
+      } else if (mimeTypeString === "image/webp") {
+        mimeType = 6;
+      } else if (mimeTypeString === "") {
+        return PromiseReject(
+          new DOMException(
+            `The MIME type of source image is not specified.
+hint: When you want to get a "Blob" from "fetch", make sure to go through a file server that returns the appropriate content-type response header,
+      and specify the URL to the file server like "await(await fetch('http://localhost:8000/sample.png').blob()".
+      Alternatively, if you are reading a local file using 'Deno.readFile' etc.,
+      set the appropriate MIME type like "new Blob([await Deno.readFile('sample.png')], { type: 'image/png' })".\n`,
```

Comment:
Suggestions may be better written in runtime/fmt_errors.rs.

Ref #26252

Reply:
Thanks.

@crowlKats 
Could I move this hint to `runtime/fmt_errors.rs`?

Reply:
Yep

Reply:
Thank you for your suggestion and permission, but I reverted this commit.
The `runtime/fmt_errors.rs` is exactly as implemented, suggesting hints from error messages in the runtime stack trace.
I think it's preferable to provide hints for specific APIs, as in this implementation, in the implementation itself from a maintenance perspective.
There are other places where hints are implemented, for example, [here](https://github.com/denoland/deno/blob/9fe52b1e8da261471a453c50a249e4e5c6ad5201/cli/graph_util.rs#L811-L813).

Regarding this issue, I think it would be better to devise a common system for displaying hints later.